### PR TITLE
Add support for insert or ignore

### PR DESCRIPTION
**Pull request by EdAyers**
Using the [sqlite definition](https://www.sqlite.org/syntax/insert-stmt.html). 

Still need to do unit tests, I'm not sure what to do if the sql dialect doesn't support `INSERT OR IGNORE`. Eg Postgres I think instead uses `ON CONFLICT DO NOTHING` after values. Maybe it's better just to not have this.

### CODE REVIEW

**Comment by tobymao** [comment]
run `make check` to do styling and tests

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -1070,17 +1070,26 @@ class Index(Expression):
         "amp": False,  # teradata
     }
 
-
 class Insert(Expression):
     arg_types = {
         "with": False,
         "this": True,
+        "or_cond": False,
```
Comment (tobymao) [inline sqlglot/expressions.py:1077]: just rename this to "or"

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -1070,17 +1070,26 @@ class Index(Expression):
...
         "expression": False,
         "overwrite": False,
         "exists": False,
         "partition": False,
     }
 
+    @property
+    def or_cond(self):
```
Comment (tobymao) [inline sqlglot/expressions.py:1085]: delete this method

File: sqlglot/generator.py [hunk 3]
```diff
@@ -932,20 +932,22 @@ def lockingproperty_sql(self, expression: exp.LockingProperty) -> str:
 
     def insert_sql(self, expression: exp.Insert) -> str:
         overwrite = expression.args.get("overwrite")
+        or_cond = "" if expression.or_cond is None else f"OR {expression.or_cond} "
```
Comment (tobymao) [inline sqlglot/generator.py:935]: self.sql(expression, "or")

File: sqlglot/tokens.py [hunk 4]
```diff
@@ -117,6 +117,7 @@ class TokenType(AutoName):
     OBJECT = auto()
 
     # keywords
+    ABORT = auto()
```
Comment (tobymao) [inline sqlglot/tokens.py:120]: remove all of these keywords that you added

File: sqlglot/parser.py [hunk 5]
```diff
@@ -1423,6 +1432,11 @@ def _parse_describe(self) -> exp.Expression:
 
         return self.expression(exp.Describe, this=this, kind=kind)
 
+    def _parse_or_cond(self):
+        if self._match_text_seq("OR"):
+          if self._match_set(self.INSERT_OR_KINDS):
```
Comment (tobymao) [inline sqlglot/parser.py:1437]: self._match_texts(...)

File: sqlglot/parser.py [hunk 6]
```diff
@@ -1423,6 +1432,11 @@ def _parse_describe(self) -> exp.Expression:
 
         return self.expression(exp.Describe, this=this, kind=kind)
 
+    def _parse_or_cond(self):
```
Comment (tobymao) [inline sqlglot/parser.py:1435]: add type -> TokenType

**Comment by GeorgeSittas** [comment]
@EdAyers any plans to finish this? Let me know if you need any help

File: sqlglot/parser.py (hunk 6 above)
Comment (EdAyers) [reply sqlglot/parser.py]: I can't figure out what I'm supposed to return here

**Comment by GeorgeSittas** [comment]
Closed in favor of #1184

**Pull request closed without merging**